### PR TITLE
Add back close checks

### PR DESCRIPTION
**Pull request by Ngoguey42**
~This doesn't seem to fix any UT~ It fixes some UTs

### CODE REVIEW

**Comment by Ngoguey42** [comment]
I could have pushed my commits to #1871. Sorry about that

File: src/irmin-pack/unix/ext.ml [hunk 1]
```diff
@@ -140,15 +145,16 @@ module Maker (Config : Conf.S) = struct
...
+              "[pack] batch failed. calling flush. (exn was %s)"
+                (Printexc.to_string exn)];
             let () =
               match File_manager.flush t.fm with
               | Ok () -> ()
               | Error _ ->
                   [%log.err
-                    "[pack] batch failed and flush failed. Silencing flush fail"];
-                  (* TODO: Proper error message (tostring on error) *)
-                  assert false
```
Comment (Ngoguey42) [inline src/irmin-pack/unix/ext.ml:151]: This `assert False` shouldn't have been there

**Comment by tomjridge** [comment]
Side comment: It's hard to tell if the code is correct. For example: maybe Inner exposes functions that need a close check, but are not patched by the Indexable.Closeable (of course, one can figure this out, but it is difficult if just locally reading the code). For this reason, I much prefer explicitly adding close checks eg `let f x y z = check_close_then (fun () -> original_f x y z)`. But this is a side issue.

We should just merge this I think.

**Comment by tomjridge** [comment]
> 

Ah, good point! I retract my previous comment!

**Pull request merged**